packages/language/test/fourslash/preprocessor/compiler-options/service.ts

```diff
@@ -0,0 +1,42 @@
+/**
+ * This program and the accompanying materials are made available under the terms of the
+ * Eclipse Public License v2.0 which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-v20.html
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Copyright Contributors to the Zowe Project.
+ *
+ */
+
+/// <reference path="../../framework.ts" />
+
+// @wrap: process
+// @deferred
+////*PROCESS <|1:NOSERVICE|>;
+////*PROCESS <|2:SERVICE|>;
+////*PROCESS <|4:SERVICE|>(<|5:)|>;
+////*PROCESS <|6:SERVICE|>(<|7:"xXx"|>);
+////*PROCESS <|8:SERVICE|>(<|9:'yYy'|>);
+////*PROCESS <|10:SERVICE|>(<|11>);
+////*PROCESS <|12:SERVICE|>(xYz);
+
+testAPI.replaceIndex(11, "x".repeat(128));
+testAPI.execute();
```

Comment:
I'm not sure this is a good idea - as in: the whole `replaceIndex`, `@deferred` and `execute` system. One of the advantages that the fourslash framework has, is that its not easily possible to write a test that acts on generated code. Instead, all the code needs to be stated explicitly in the test document. This `replaceIndex` method allows to circumvent this completely.

If you need a test that generates code, please don't use the fourslash framework to do this. There are also a few other issues related to this (i.e. tokenization eventually becoming async, which will make `execute` async as well, etc.), which will make the testing framework much harder to maintain in the future.